enhancements/multi-arch/heterogeneous-architecture-clusters.md

```diff
@@ -0,0 +1,450 @@
+- Testing scenarios would include machine set scaling, upgrades involving ARM nodes and OVN testing on heterogeneous clusters.
+- New release payload acceptance jobs on heterogeneous clusters.
+
+Testing for Phase 2:
+- With the addition of Power and Z architectures, there will be more permutations to test. We would need to figure out the absolute essential set of
+permutations and the tests to run on those.
+- Imagestream conformance tests.
+- OLM and console testing
+- Test migration and upgrade scenarios
+
+### Graduation Criteria
+
+Phase 1 will be a 4.11 techpreview with documented limitations. The techpreview no upgrade feature gate would be set and upgrades would have to be forced
+manually.
+
+#### Dev Preview -> Tech Preview
+
+There will be a tech preview of basic functionality (Phase 1 as described above) in 4.11
+
+#### Tech Preview -> GA
+
+TBD
+
+#### Removing a deprecated feature
+
+This enhancement does not describe removing a feature of Openshift, this section is not applicable.
+
+### Upgrade / Downgrade Strategy
+
+The upgrade strategy is described in the design details section above. The important changes here are having a separate architecture
+for the heterogeneous releases and also the work needed to support upgrades from a homogeneous release -> heterogeneous release in
+the future.
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+- The ClusterVersion API's spec field will have an additional architecture field (to be appropriately named) which would serve as an indicator for
+whether the cluster is homogeneous or heterogeneous. The ClusterVersionStatus API will also have a similar field.
+- The NodePoolSpec API in Hypershift would need to have an architecture field so it can pick the images corresponding to the relevant architecture
+- The imagestreams API would also undergo changes as described in the imagestreams section.
+
+#### Failure Modes
+
+No new failure modes are introduced as part of this feature.
+
+#### Support Procedures
+
+TBD
+
+## Implementation History
+
+A heterogeneous payload was built as a proof of concept and spun up worker nodes using ARM with an x86 control plane on AWS. It worked,
+but here are the OCP features we know won’t work properly in such an environment today which are explained in detail above:
+- Imagestreams - does not fully support manifestlists, they will import a single architecture (by default based on the control plane’s architecture,
+though it can be chosen explicitly) from the manifestlist, so anything referencing an imagestream tag will get the image associated with that
+architecture, regardless of what architecture the node is running.
+- Internal registry - does not support manifestlists (storing or pulling). This also means you cannot use the internal registry for image
+pull-through (proxying) for manifestlist-based content.
```

Comment:
these two bullets (imagestreams, registry) no longer reflect the plan?

Reply:
updated.